internal/util/cache/lru_cache.go

```diff
@@ -164,49 +192,51 @@ func (c *LRU) Keys() []Key {
 	keys := make([]Key, len(c.items))
 	i := 0
 	for ent := c.evictList.Back(); ent != nil; ent = ent.Prev() {
-		keys[i] = ent.Value.(*entry).key
+		keys[i] = ent.Value.(*Entry).Key
 		i++
 	}
 	return keys
 }
 
-func (c *LRU) Len() int {
+func (c *LRU) Size() uint64 {
 	c.m.RLock()
 	defer c.m.RUnlock()
-	return c.evictList.Len()
+	return c.size
 }
 
-func (c *LRU) Capacity() int {
+func (c *LRU) Capacity() uint64 {
 	return c.capacity
 }
 
 func (c *LRU) Purge() {
 	c.m.Lock()
```

Comment:
How do we handle if evictedCh full? 
If purge we should handle evicthere but not send into evictedCh?

Reply:
lgtm